File: casa_guido/app/src/main/java/br/com/casa_guido/models/ComposicaoFamiliar.kt
package br.com.casa_guido.models

import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient
import java.util.UUID

@Serializable
data class ComposicaoFamiliar(
    @Transient
    val id: String = UUID.randomUUID().toString(),
    val nome: String = "",
    val parentesco: String = "",
    val dataNascimento: String = "",
    val idade: Int = 0,
    val estudaOpcional: Int = 0,
    val serie: String = "",
    val trabalhaOpcional: Int = 0,
    val renda: String = "",
)

@Serializable
data class ComposicaoFamiliarDTO(
    val nome: String = "",
    val parentesco: String = "",
    val dataNascimento: String = "",
    val idade: Int = 0,
    val estudaOpcional: Int = 0,
    val serie: String = "",
    val trabalhaOpcional: Int = 0,
    val renda: String = "",
)

fun ComposicaoFamiliar.toRequestDTO(): ComposicaoFamiliarDTO {
    return ComposicaoFamiliarDTO(
        nome = this.nome,
        parentesco = this.parentesco,
        dataNascimento = this.dataNascimento,
        idade = this.idade,
        estudaOpcional = this.estudaOpcional,
        serie = this.serie,
        trabalhaOpcional = this.trabalhaOpcional,
        renda = this.renda
    )
}

File: backend/models/ComposicaoFamiliar.js
const mongoose = require('mongoose');

const ComposicaoFamiliarSchema = new mongoose.Schema({
  paciente: { type: mongoose.Schema.Types.ObjectId, ref: 'Paciente' },
  nome: String,
  parentesco: String,
  dataNascimento: String,
  estudaOpcional: Number,
  serie: String,
  trabalhaOpcional: Number,
  renda: String
});

module.exports = mongoose.model('ComposicaoFamiliar', ComposicaoFamiliarSchema); 